### DISCUSSION TITLE
Regression in Docusaurus v3 with regard to rendering of Markdown with CJK text

### DISCUSSION THREAD

**Discussion by allenscha** [facebook/docusaurus#9644] [Q&A] [answered]
In Docusaurus v3, text including CJK text along with parentheses and bold style (or any other basic Markdown style) is not rendered as expected.

For example, for the following text:

```
Let's look at the definition of **database management system (DBMS)**.

**DBMS(데이터베이스 관리 시스템)**의 정의를 알아봅시다.

**DBMS（データベース管理システム）**の定義を見てみましょう。

让我们了解一下**DBMS（数据库管理系统）**的定义。
```

In the rendered result, the bold style is not applied as expected for text including CJK text.

> Let's look at the definition of **database management system (DBMS)**.
> 
> **DBMS(데이터베이스 관리 시스템)**의 정의를 알아봅시다.
> 
> **DBMS（データベース管理システム）**の定義を見てみましょう。
> 
> 让我们了解一下**DBMS（数据库管理系统）**的定义。

(I was surprise to see that this issue is reproduced in the GitHub Markdown editor.)

- This issue did not exist in Docusaurus v2.
- The [MDX playground](https://mdxjs.com/playground/) shows the same issue.

Any good idea on how to solve this issue?

**Comment by Josh-Cena** [comment, marked as answer]
Yes, if GitHub and MDX both have this issue, that means there's a very high chance that it's how GitHub-flavored Markdown is supposed to work. MDX v2 changes parsing behavior from MDX v1, so perhaps they think this is more correct.

If you wish, maybe add spaces?

```md
让我们了解一下 **DBMS（数据库管理系统）** 的定义。
```

I know it looks awkward but it is what it takes to make things work. Otherwise you would need to use HTML tags like `<strong>`.

In any case, it would be better to ask in the MDX repo instead since we don't control Markdown parsing.

**Comment by allenscha** [comment]
Adding spaces for these types of construct is typographically unacceptable for official documentation in CJK, so we'll have to use corresponding HTML tags instead, until the issue is resolved.

I'll see what I can do in the MDX repo. Thank you.

**Reply by rxliuli** [reply]
The best approach is to wait for the standard's support, as referenced here: <https://github.com/commonmark/commonmark-spec/issues/650>. However, the real world cannot wait too long, so you can use some plugins to automatically remove extra spaces when rendering markdown to HTML, or write your own plugins for parsing.

I have previously written related plugins for remark/markdown-it, which have been used to render several novels. You can refer to them.

- remark: https://github.com/rxliuli/liuli-tools/blob/82d81eeb0d661ad1338d04d1a75f173764dea9bc/packages/markdown-util/src/cjk.ts#L83-L109
- markdown-it: https://github.com/mark-magic/mark-magic/blob/77f9b1571a7d96847fc39e0aa8504ed994a64b71/packages/plugin-docs/src/assets/config.ts#L13-L42

Originally, `**真，** 她` would render as `<p><strong>真，</strong> 她</p>`, but after plugin processing, it becomes `<p><strong>真，</strong>她</p>`.

Another direction is parsing, which is very complex and requires changes to the markdown specification to resolve. However, plugins can still partially solve this. I am currently writing and testing plugins for remark, which you can refer to: https://github.com/rxliuli/liuli-tools/blob/82d81eeb0d661ad1338d04d1a75f173764dea9bc/packages/markdown-util/src/__tests__/cjk.test.ts#L6-L48

**Reply by allenscha** [reply]
Thank you for the pointer to the CommonMark issue page. This looks way more complicated that it seemed at first...
